src/contracts/fungible/cache/file.rs

```diff
@@ -192,4 +192,34 @@ impl Cache for FileCache {
         self.save()?;
         Ok(existed)
     }
+
+    fn utxo_allocation_map(
+        &self,
+        contract_id: ContractId,
+    ) -> Result<BTreeMap<bitcoin::OutPoint, AtomicValue>, CacheError> {
+        let asset = self.asset(contract_id).unwrap();
+
+        let allocation_map = asset.known_allocations();
+
+        let mut result = BTreeMap::new();
+
+        for item in allocation_map {
+            let mut sum = 0;
+            for alloc in item.1 {
+                sum += alloc.value().value;
+            }
+
+            result.insert(item.0.clone(), sum);
```

Comment:
This replaces any previous assignments to the same outpoint if they were existing, leading to incorrect return data. See my commend above about replacing the value with the `Vec`